server/service.go

```diff
@@ -416,8 +454,13 @@ func (m *BoostService) handleGetPayload(w http.ResponseWriter, req *http.Request
 	// Wait for all requests to complete...
 	wg.Wait()
 
+	// If no payload has been received from relay, log loudly about withholding!
 	if result.Data == nil || result.Data.BlockHash == nilHash {
-		log.Warn("getPayload: no valid response from relay")
+		bidKey := bidRespKey{slot: payload.Message.Slot, blockHash: payload.Message.Body.ExecutionPayloadHeader.BlockHash.String()}
+		m.bidsLock.Lock()
```

Comment:
Q: can we respond before locking and logging?

Reply:
yes!

Reply:
this is sadly only going to work for getPayload, not getHeader